### ISSUE TITLE
Autoimports don't work in global settings with new architecture

### ISSUE THREAD

**Issue by Blaisorblade** [sbt/sbt#2120]
See typesafehub/sbteclipse#295 and sbt/sbt-pgp#80.

> I have installed sbteclipse as a global plugin (since in most of my builds nothing depends on it). I have some configuration for it in ~/.sbt/0.13/sbteclipse.sbt, that uses EclipseKeys unqualified. When I switched from 2.5.0 to 4.0.0, I needed to add an import there for EclipseKeys. This is not needed if I place the setting in build.sbt, whether the plugin is installed locally or globally.

For sbteclipse, it seems that's only needed with versions of the plugin after switching to the 0.13.5 plugin architecture — that is, after https://github.com/typesafehub/sbteclipse/commit/b6b0379ef9b9c9602a77b7dc09fb1c2bf25b8610#diff-e127bb09dd9d0b8de01cd7be195d59bcR30. This is just a first hypothesis, since I haven't rebuilt a single line of code. But anyhow, with the info I have this looks like a SBT bug.

**Comment by Blaisorblade** [comment]
> For sbteclipse, it seems that's only needed with versions of the plugin after switching to the 0.13.5 plugin architecture

That's also plausible for sbt-pgp ([first offending commit](https://github.com/sbt/sbt-pgp/commit/4f1f80d23eaa413c178d1518a59e1f80cfecefdf)).

In either case, I haven't yet bisected the plugin to confirm my hypothesis. I have just asked the following question: "Between the working and non-working version, did this plugin change the way it tells SBT which names to autoimport in configuration?", and the answer seems, in both cases, "yes, they switched to a newer API".

**Comment by eed3si9n** [comment]
I've converted a few of my plugins (like [sbt-assembly](https://github.com/sbt/sbt-assembly/commit/331f3fdabb58b5350d6a6e26bbccf022d9de0942)) and others (like [bintray-sbt](https://github.com/softprops/bintray-sbt/pull/53)) and thus far I have not encountered this issue.
I've also tried to adhere to [Plugins Best Practices](http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html) strictly.

Is there something peculiar that sbteclipse and sbt-pgp is doing? /cc @jsuereth

**Comment by timcharper** [comment]
It seems like what is happening is that the imports for these keys is happening too late.

If I load the plugin via a file in `~/.sbt/0.13/plugins/gpg.sbt`:

``` scala
addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
```

And then put my configuration in  `~/.sbt/0.13/gpg.sbt`:

``` scala
pgpPassphrase := Some("<redacted>".toCharArray)

pgpSecretRing := file("<redacted>/secring.asc")

pgpPublicRing := file("<redacted>/pubring.asc")
```

... then I get the following error:

```
/Users/timcharper/.sbt/0.13/gpg.sbt:3: error: not found: value pgpPassphrase
pgpPassphrase := Some("<redacted>".toCharArray)
```

However, if comment out the contents of `~/.sbt/0.13/gpg.sbt`, and instead place the configuration in my project `build.sbt`, then I no longer get the error.

So, `AutoImport` is working, but it seems like the load order is as follows:
- load `~/.sbt/0.13/plugins/gpg.sbt`
- load `~/.sbt/0.13/gpg.sbt`
- `import com.typesafe.sbt.pgp.PgpKeys._`
- load `./build.sbt`

**Comment by eed3si9n** [comment]
@timcharper Interesting observation. That could be it.

**Comment by eed3si9n** [comment]
Note to someone who wants to try fixing this https://github.com/sbt/sbt/blob/v0.13.9/main/src/main/scala/sbt/Load.scala#L74-L85

**Comment by timcharper** [comment]
As seen ^^, I've a fix submitted as a pull request for this issue.

**Comment by dwijnand** [comment]
Fixed in #2399.
